install_config/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,427 @@
+  # Subject Alternative Name in the output of:
+  # openssl x509 -text -in /etc/pki/tls/certs/proxy.example.com.crt
+  ServerName proxy.example.com
+
+  DocumentRoot /var/www/html
+  SSLEngine on
+  SSLCertificateFile /etc/pki/tls/certs/proxy.example.com.crt
+  SSLCertificateKeyFile /etc/pki/tls/private/proxy.example.com.key
+  SSLCACertificateFile /etc/pki/CA/certs/ca.crt
+
+  # Send logs to a specific location to make them easier to find
+  ErrorLog logs/proxy_error_log
+  TransferLog logs/proxy_access_log
+  LogLevel warn
+  SSLProxyEngine on
+  SSLProxyCACertificateFile /etc/pki/CA/certs/ca.crt
+  # It's critical to enforce client certificates on the Master.  Otherwise
+  # requests could spoof the X-Remote-User header by accessing the Master's
+  # /oauth/authorize endpoint directly.
+  SSLProxyMachineCertificateFile /etc/pki/tls/certs/authproxy.pem
+
+  # Send all requests to the console
+  RewriteEngine              On
+  RewriteRule     ^/console(.*)$     https://%{HTTP_HOST}:8443/console$1 [R,L]
+
+  # In order to using the challenging-proxy an X-Csrf-Token must be present.
+  RewriteCond %{REQUEST_URI} ^/challenging-proxy
+  RewriteCond %{HTTP:X-Csrf-Token} ^$ [NC]
+  RewriteRule ^.* - [F,L]
+
+  <Location /challenging-proxy/oauth/authorize>
+    # Insert your backend server name/ip here.
+    ProxyPass https://openshift.example.com:8443/oauth/authorize
+    AuthType Basic
+    AuthBasicProvider PAM
+    AuthPAMService openshift
+  </Location>
+
+  <ProxyMatch /oauth/authorize>
+    AuthName openshift
+    Require valid-user
+    RequestHeader set X-Remote-User %{REMOTE_USER}s
+  </ProxyMatch>
+</VirtualHost>
+
+RequestHeader unset X-Remote-User
+----
+====
+
+Set a boolean to tell SELinux that it is acceptable for Apache to contact the
+PAM subsystem:
+
+====
+----
+# setsebool -P allow_httpd_mod_auth_pam on
+# setsebool -P httpd_dbus_sssd on
+----
+====
+
+Start up Apache:
+
+====
+----
+# systemctl start httpd.service
+----
+====
+
+[[sssd-phase-3-openshift-configuration]]
+== Phase 3: {product-title} Configuration
+
+This section describes how to set up an {product-title} server from scratch in
+an "all in one" configuration.
+link:../install_config/master_node_configuration.html[Master and Node
+Configuration] provides more information on alternate configurations.
+
+First, modify the default configuration to use the new identity provider just
+created. Start by modifying the *_/etc/origin/master/master-config.yaml_* file.
+Scan through it and locate the *identityProviders* section and replace it with:
+
+====
+----
+  identityProviders:
+  - name: any_provider_name
+    challenge: true
+    login: false
+    mappingMethod: claim
+    provider:
+      apiVersion: v1
+      kind: RequestHeaderIdentityProvider
+      challengeURL: "https://proxy.example.com/challenging-proxy/oauth/authorize?${query}"
+      clientCA: /etc/origin/master/proxy/proxyca.crt
+      headers:
+      - X-Remote-User
```

Comment:
See https://sgallagh.wordpress.com/2016/05/27/openshift-and-sssd-part-3-extended-ldap-attributes/ for details on how to add the email and full-name attributes.